lib/html2text.rb

```diff
@@ -24,18 +24,20 @@ def self.replace_entities(text)
   end
 
   def convert
-    output = iterate_over(doc)
-    output = remove_leading_and_trailing_whitespace(output)
-    output = remove_unnecessary_empty_lines(output)
+    output = process(doc)
+    remove_leading_and_trailing_whitespace(output)
+    remove_unnecessary_empty_lines(output)
     output.strip
   end
 
   def remove_leading_and_trailing_whitespace(text)
-    text.gsub(/[ \t]*\n[ \t]*/im, "\n").gsub(/ *\t */im, "\t")
+    # String#gsub! returns nil if no substition was performed, so these calls can't be chained.
+    text.gsub!(/[ \t]*\n[ \t]*/im, "\n")
+    text.gsub!(/ *\t */im, "\t")
```

Comment:
`remove_leading_and_trailing_whitespace` and `remove_unnecessary_empty_lines` will now sometimes return `nil` instead of returning the `text`. Is that ok? Seems like we could keep `gsub!` for the memory benefits and still preserve the existing behavior with an explicit return.

Reply:
Pushed a commit with that change